sdks/python/apache_beam/io/avroio_it_test.py

```diff
@@ -0,0 +1,237 @@
+
+    multiplier = 10
+    assert (multiplier % 2 == 0), \
+      "Use an even multiplier to not compare floats."
+
+    # A `PCollection` with `num_records` avro records
+    _ = (write_pipeline
+         | 'CreateBatches' >> Create(batches)
+         | 'ExpandBatches' >> FlatMap(batch_indices)
+         | 'Multiply' >> Map(lambda x: multiplier * x)
+         | 'CreateRecords' >> Map(record)
+         | 'WriteAvro' >> WriteToAvro(files_output,
+                                      self.SCHEMA,
+                                      file_name_suffix=".avro",
+                                      use_fastavro=self.use_fastavro)
+        )
+
+    write_result = write_pipeline.run()
+    write_result.wait_until_finish()
+    assert write_result.state == PipelineState.DONE
+
+    def check(x, mean):
+      if not x == mean:
+        raise BeamAssertException('Assertion failed: %s == %s' % (x, mean))
+
+    def check_distr(lst):
+      if not len(set(lst)) == 1:
+        raise BeamAssertException(
+            'Labels should be equally distributed: %s ' % (lst))
+
+    def check_concat(concatenated_string):
+      split_concatenated_string = concatenated_string.split(" ")
+      for i in range(len(batches) * batch_size):
+        if str(multiplier * i) not in split_concatenated_string:
+          raise BeamAssertException(
+              'String should be in concatenation: %s' % str(multiplier * i))
+
+    def check_for_none(lst):
+      if not None in lst:
+        raise BeamAssertException('None value was not preserved: %s ' % (lst))
+
+    def concat_strings(values):
+      return " ".join(values)
+
+    read_pcol = (read_pipeline
+                 | 'ReadAvro' >> ReadFromAvro(files_output + '*')
+                )
+
+    _ = (read_pcol
+         | 'ExtractLabel' >> Map(lambda x: x['label'])
+         | 'MakeLabelsKV' >> Map(lambda x: (x, 1))
+         | 'CountLabels' >> Count.PerKey()
+         | 'ExtractLabelCounts' >> Map(lambda x: x[1])
+         | 'LabelsToList' >> ToList()
+         | 'CheckEqualDistributionLabels' >> Map(lambda x: check_distr(x))
+        )
+
+    _ = (read_pcol
+         | 'ExtractNumber' >> Map(lambda x: x['number'])
+         | 'CalculateMean' >> Mean.Globally()
+         | 'MeanIsCorrect' >> Map(
+             lambda x: check(x, multiplier *
+                             ((len(batches) * batch_size) - 1)/2))
+        )
+
+    _ = (read_pcol
+         | 'ExtractNumberStr' >> Map(lambda x: x['number_str'])
+         | 'Concatenate' >> CombineGlobally(concat_strings)
+         | 'CheckConcatenation' >> Map(lambda x: check_concat(x))
+        )
+
+    _ = (read_pcol
+         | 'ExtractNumberColor' >> Map(lambda x: x['color'])
+         | 'MakeColorsKV' >> Map(lambda x: (x, 1))
+         | 'CountColors' >> Count.PerKey()
+         | 'ExtractColorCounts' >> Map(lambda x: x[0])
+         | 'ColorsToList' >> ToList()
+         | 'CheckPreservationOfNone' >> Map(lambda x: check_for_none(x))
```

Comment:
Old comment valentyn: Not sure I understand Line 208 - what are we checking here and how?
Resolution: Changed naming